### ISSUE TITLE
Auto upload not triggered

### ISSUE THREAD

**Issue by christianlupus** [nextcloud/android#9320]
### Steps to reproduce
1. Set up the app to do auto uploads (e.g. use camera `DCIM` folder as suggested by default)
2. (Optionally) Empty list of uploads to have a good view on the newly created (auto) uploads
3. Take an image (which should trigger the upload)
4. Wait a few minutes for the app to detect the newly created file

### Expected behaviour
After some time (in the range of 5-10 min) the app should at least register an upload in the uploads section. Depending on the settings (only via WiFi for example), the upload will take place immediately, be queued, or be stalled.

### Actual behaviour
The upload is never queued. It should have created anything in the list (be it stalled, queued, in progress, or done in the meantime). In fact, there is nothing logged at all.

I suspect that the detection of new files somehow fails.

### Can you reproduce this problem on https://try.nextcloud.com?
As no upload is triggered, no network is involved. => Not tested

### Environment data
Android version: 10

Device model: Dogee S96 Pro

Stock or customized system: Stock

Nextcloud app version: 3.18.0 RC6

Nextcloud server version: 22.2.3

Reverse proxy: Apache

#### Web server error log
Not network related

#### Nextcloud log (data/nextcloud.log)
Not network related

#### App logs
When making a photo and looking at the logs within the app, there is no line logged within a few minutes after the photo.

Unfortunately, the logs only go back until today morning, so older logs seem to have been rolled out. I cannot compare with a time where things were working.

**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

### Additional information

This seems to be a regression, as it used to work until recently. I was on holiday and during part of it the auto uploads were still working as intended. Sometime around Nov 06th it suddenly stopped working. I am not sure about the exact point in time as my network connection was a bit hairy and I was not always able to use the NC server accordingly. So there was build a big list of outstanding uploads over these days. At some point, this list just did not get any longer.

**Comment by danielroehrig** [comment]
My upload queue has been empty for 3 weeks now. Neither photos nor videos are getting picked up. If I create a new custom folder, these files will be uploaded eventually, but the camera photo/video folder seems to be ignored. It used to work flawlessly until 3 weeks ago. Things I tried:

1. Disabling energy saving mode for nextcloud
2. Updating nextcloud app to version 3.18 RC6
3. Renaming the target folder on the server (so that it should see all files as missing as the folder is not there)
4. Clearing cache, restarting the phone
5. Checking, if the upload functionality on a different account on the same server with a different phone still works. It does.

Here some (maybe) interesting logs:
```
2021-11-20T10:05:45.613+0100;D;UploadsStorageManager;QUERY: (status== ? OR last_result==9 OR last_result==13 OR last_result==11 OR last_result==14 ) AND last_result!= 16 ROWID: -1
2021-11-20T10:05:45.616+0100;V;UploadsStorageManager;getUploads() got 0 rows from page 0, 0 rows total so far, last ID -1
2021-11-20T10:05:45.616+0100;V;UploadsStorageManager;getUploads() returning 0 (0) rows after reading 1 pages
---
2021-11-20T10:05:45.769+0100;V;ArbitraryDataProvider;Updating arbitrary data with cloud id: global key: media_folders value: {"imageMediaFolders":[[...]"/storage/emulated/0/DCIM/Camera",[...]],"videoMediaFolders":[]}
```
So, for me it seems like it looks at the right place but doesn't find any new photos or videos despite there being like 40 or 50.

**Comment by ksmolder** [comment]
I'm facing the same issue with Nextcloud client 3.18 for Android an a OnePlus Nord. I'm not sure however if the phone's make and model is of any importance here...
During the client setup, I did notice something strange. When I select the folder to watch (example `/storage/emulated/0/DCIM/Camera`), the folder displayed as the "source" folder reads `/storage/emulated/0/**DCIM**`. So it seems as if the parent folder of the selected folder is watched?

**Comment by AlvaroBrey** [comment]
@ksmolder thanks for the tip; this might be a consequence of #8940 In the meantime try setting up the folder with the workaround there

**Comment by gjedeer** [comment]
For me, the last uploaded file was from November 14, no photos taken later than that have been uploaded. `/storage/emulated/0/DCIM/Camera` is the path in settings. Pixel phone recently updated to Android 12 - but not sure if the date of Android update was November 14.

**Comment by martinmoreira** [comment]
Hello.
Not sure if you managed to solve this, but for me denying and re-allowing the storage permissions fixed this issue.

**Comment by danielroehrig** [comment]
@martinmoreira Could you elaborate? How did you do this?

**Comment by martinmoreira** [comment]
Of course.
For me it is:
Android Settings / Apps & Notifications / See all Apps / Nextcloud / Permissions / Storage
Then choose "Deny"
And the "Allow again".

**Comment by danielroehrig** [comment]
I can't believe that actually worked! Thanks @martinmoreira

**Comment by christianlupus** [comment]
For me, this workaround did work. So far I am not affected anymore.
However, I do not know what triggered the problem and thus I am not sure if it will happen again. Also I do not know how I could get notified/aware of it happens again.

**Comment by rarealphacat** [comment]
It does not work for me as there is no "Storage" permission to alter between deny and allow. I'm on Android 11, OnePlus 6, OxygenOS 11.1.2.2

**Comment by cfelicio** [comment]
Looks like auto upload also stopped working on my android phone right after the 3.18 release (the date of the last uploaded picture coincides with the release of the version). My other android phone still running 3.15 continues to work fine. I will try to remove / reinstall / reconfigure 3.18 and see if it works again.

**Comment by IPlayZed** [comment]
> Of course. For me it is: Android Settings / Apps & Notifications / See all Apps / Nextcloud / Permissions / Storage Then choose "Deny" And the "Allow again".

This is not possible on Android 11, OneUI 4.0,  Samsung S21 Ultra.

Does _anybody have any idea what is causing this?_

**Comment by NicolasGoeddel** [comment]
> This is not possible on Android 11, OneUI 4.0, Samsung S21 Ultra.

I do not own a Samsung S21 Ultra but after a quick google search you should find it in Settings -> Privacy -> Permission Manager -> Storage

**Comment by IPlayZed** [comment]
Ok, so appareantly I took a picture with my phone and it triggered the upload sequence for _every_ upload folder. Maybe the problem lies in it not being triggered by anything?

**Comment by IPlayZed** [comment]
> > This is not possible on Android 11, OneUI 4.0, Samsung S21 Ultra.
> 
> I do not own a Samsung S21 Ultra but after a quick google search you should find it in Settings -> Privacy -> Permission Manager -> Storage

Nextcloud is not there...

However it is in Settings > Apps > Special access > All files access.
For interest: I also found that on this setup you have to manually go and enable sync in system settings:  Settings > Accounts and backup  > Manage accounts > Nextcloud
Idk if this makes a difference but shouldn't this be enabled by default.

Still, there must be some deep rooted problem with the autosync engine because it is not working on mobile internet connection, is only triggered by new items, etc.

**Comment by sanchosk** [comment]
Is it possible this is somehow "feature" of android instead of Nextcloud problem? It looks to me as if OS decided that I haven't opened the nextcloud client for a long time, so the permission was removed, even though it's shown as allowed.
Not sure if this is it or, perhaps, this is caused by OS update...
My latest OS update was sometime at start of April which is also the last time my auto upload worked.
Once I removed and re-applied the permission (thanks, @martinmoreira !!!) it works flawlessly again.

**Comment by IPlayZed** [comment]
@sanchosk in newer Android OSes there is a feature where app permissions, if not used, are disabled. That is a good feature actually, but I am not talking about that. Nextcloud should set these options ON by default during first launch or should periodically check against them.
Also, there is the seperate question of when and how uploads are triggered.

**Comment by maisim** [comment]
> Hello. Not sure if you managed to solve this, but for me denying and re-allowing the storage permissions fixed this issue.

I can confirm this.
I just discover auto upload stop working since months.
When I looked at android logs, nothing was happening when a new photo was taken.

As @martinmoreira suggests, disable storage permission for nextcloud app and re-allowing it immediately make uploads start immediately

**Comment by b3** [comment]
On my Fairphone 2 running /e/ OS 1.0 (which is Android 10 on my phone) this workaround also worked but I had to 

1. start settings
2. disable the storage permission
3. leave settings
4. wait a bit 15/20s
5. start settings
6. enable storage permission

This is still just a **workaround** not a bug fix. 

I am still curious to get why this permission seems no more effective (does it come from Android or from Nextcloud).

**Comment by rarealphacat** [comment]
I manually upload some newly taken picture files and that might trigger the whole auto upload process. Video files works but the auto upload process takes a bit longer to begin (~10mins). Maybe because phone still processing?

**Comment by azm-technik** [comment]
I'm always having troubles with background activities of different apps when they are not using Google Play Services. Looks like everything not Google is a third class citizen in Android.

**Comment by www-flomei-de** [comment]
Removing storage permission and then granting it again, also fixed an auto upload issue on a Huawei Mate 20 Pro with up to date system.

Also removing and regranting those permissions led to Nextcloud discovering lots of new folders for auto upload.

**Comment by rcsilva83** [comment]
@www-flomei-de this also worked for me on my Xiaomi Poco F1! Thanks a lot!

**Comment by benibr** [comment]
Thanks for the workaround of reallowing the storage permission, I was going mad.

I can confirm, that with very large folders, the autouploading does not start instantly but after some minutes is did.

**Comment by drumstick77** [comment]
This still solution still doesn't work for me guys. I'm on Samsung s22 ultra. It was working on my old server that went kaputz. So I've rebuilt it, for my laptop sync'd again successfully, but Android still says no. Any ideas? Thanks

**Comment by nZeus** [comment]
Still doesn't work.
Could you please add some information to the page with the status? E.g. `Pending permissions`, `Indexing`, etc?
This issue is so frustrating. Actually, this is the _core_ feature, and it doesn't work.

**Comment by drumstick77** [comment]
Uploading files manually works, but configuring it with auto upload doesn't even after trying the following:

- set chmod read write access to data folder
- set samsung to enable sync
- set samsung not to let this app to sleep
- set samsung to have full access to directories on the phone.
- I've signed in with admin account from my phone, so given the admin access, it should just work.
I can't see anything in the logs, it's very frustrating.

**Comment by thiscantbeserious** [comment]
This is an issue with the App.

You can circumvent this by using FolderSync from the Play-Store which has Nextcloud-Support and syncs instantly without any issues.

**Comment by onatbas** [comment]
This started happening since a week ago on my iPhone. I changed the server device from one to another; could this have affected it?

**Comment by 9ycbgf0k8fpg** [comment]
I had the same issue, with no new pictures detected by the Nextcloud App since a few weeks. Disabling and re-enabling storage permission did solve my issue.

**Comment by haarp** [comment]
Re-applying storage permissions also worked here. Additionally, afte doing so, many more objects are showing up in Settings->Auto upload. It seems whatever got broken prevented it from picking up (new) files.

**Comment by rcsilva83** [comment]
When I started using Nextcloud, the last thing I'd think it was going to be problematic was the file storage and sync features. It's kinda disappointing it is bugged and for so long. Unfortunately I have a feeling about all Nextcloud's modules that bug reports don't get much attention although new versions with new features are released often.

**Comment by sanchosk** [comment]
I am not sure this frustration is properly targeted.
It seems to me that Android OS removes the privileges silently.
The new version of /e/ OS now adds option for Android to remove unused privileges from apps.
Once this is unchecked, the app will keep its rights forever - just testing mine nextcloud to check if it will run into the same problem - my only worry is that I now need to NOT turn on my nextcloud app for some time ;)

**Comment by rcsilva83** [comment]
I don't think this is what is happening because the workaround is revoke the permissions and then grant them again. So the permissions weren't removed by the OS.

Additionally, Android only revokes permissions from unused apps. If Nextcloud is syncing in background all the time, it being used. That's why the OS doesn't revoke permissions from Google Photos, for example.

**Comment by meldarionqeusse** [comment]
I have been using auto upload without any problems cor the past 2.5 urs on 4 different devices, until today when my main phone refuses to sync after I had to uninstall nextcloud and reinstall it.

When I went to the auto upload feature the app crashed and gave this error well part of it cause it is very long and would not let me paste all of it here. I figured it was permission so I went to setting gave storage permission to nextcloud and I was able yo setup auto upload but so far (3hours later) nothing has started uploading not even when I take new photos.

I am running NC 25.0.2 and Android 9. I tried NC android app from googleplay and latest version downloading it directly from github, tried removing storage permission and re-enabling them. I will let it be over night see if it fixes itself

```java
Exception in thread "DefaultDispatcher-worker-1" java.lang.StackOverflowError: stack size 1041KB
    at com.owncloud.android.utils.PermissionUtil.requestExternalStoragePermission(PermissionUtil.kt:123)
    at com.owncloud.android.ui.activity.SyncedFoldersActivity.onRequestPermissionsResult(SyncedFoldersActivity.kt:781)
    at android.app.Activity.requestPermissions(Activity.java:4477)
    at androidx.core.app.ActivityCompat$Api23Impl.requestPermissions(ActivityCompat.java:850)
    at androidx.core.app.ActivityCompat.requestPermissions(ActivityCompat.java:520)
    at com.owncloud.android.utils.PermissionUtil.requestStoragePermission$doRequest(PermissionUtil.kt:163)
    at com.owncloud.android.utils.PermissionUtil.requestStoragePermission(PermissionUtil.kt:188)
    at com.owncloud.android.utils.PermissionUtil.requestExternalStoragePermission(PermissionUtil.kt:139)
    at com.owncloud.android.ui.activity.SyncedFoldersActivity.onRequestPermissionsResult(SyncedFoldersActivity.kt:781)
    at android.app.Activity.requestPermissions(Activity.java:4477)
    at androidx.core.app.ActivityCompat$Api23Impl.requestPermissions(ActivityCompat.java:850)
```

**Comment by Darkmagister** [comment]
i'm having the same issue here (app 3.23.1 android 10, NC 25.0.3) i have configured the camera folder (in nc app i see 2 folder one for video and the same one for photo but i need to configure 2 autoupload) then i removed the permission added again but only the folder for the video has been synced (maybe the number of photo is too high??) i tried to removed the autoupload from the video and leave only the photo, remove the permission and add it again and nothing...

**Comment by danieldietsch** [comment]
I have the same problem (App 3.23.1, Nextcloud 25.0.3, Android 13). Auto-upload worked fine, but suddenly stopped working one week ago and so far, none of the workarounds linked here helped restore the function. 
I will try and reinstall the app, but it just seems as if the background service is not even started (I had autoupload configured for camera images and videos as well as for a couple of apps -- none of them sync).

**Comment by meldarionqeusse** [comment]
is your camera folder on an SD card by any chance?

I had the same problem last month the last time I posted. I suspect it was an issue related to sd card permissions or something weird happening with android not granting permission for the sd card cause other folder on the internal memory were auto uploading. 

I resolved it by installing an new sd card setup auto-upload with the new sd card took a photo to ensure it would upload. Then I Removed the new sd card and re-inserted the old one, setup auto-upload again and it started working. I am still uncertain why it stopped working and why it would not start.

**Comment by danieldietsch** [comment]
No, no SD card.

**Comment by pdmtt** [comment]
Happened to me too. Autoupload had been working until it wasn't. Can't say what happened. 
Upload was not being triggered for folders configured before the failure. Setting a new folder to be autouploaded triggered the upload just for it.

Unsetting and resetting access to files workaround has not worked.

Using Motorola Moto G100 with Nextcloud v3.24.1,

**Comment by theaddies** [comment]
Same for me.  Worked great up until a few weeks ago.  I have tried all sorts of things and nothing works.  No photos are uploaded.
Samsung Galaxy A53 and nextcloud android app 3.24.2

**Comment by drumstick77** [comment]
Did you kick off any updates? Or got kicked off by accident?

**Comment by theaddies** [comment]
I switched to the beta version and it started working!  Wonderful.

**Comment by biva** [comment]
Hello @AndyScherzinger and @tobiasKaminsky , I also have this issue, and I'd like to see how we can help, as users, to solve it? (sorry, I have no coding skills :( )
Maybe we could first merge some issues (this one and https://github.com/nextcloud/android/issues/8285 for example?)
With all the people reporting issues here, would you have questions that would help you solve this issue?

For my own experience, it worked before, but then I saw that the app was uploading everything again, so I did a fresh install on the phone and the auto upload doesn't work anymore. I tried all the workarounds and solutions proposed in different bug reports (disable/enable permission - set up an other folder - try it on a folder with a low number of pictures - enabled sync in Android parameters - disable "sync existing files" - ...)

Thanks for your support!

**Comment by 007hacky007** [comment]
I'm new Nextcloud user and auto upload does not work for me either. I've tried latest Google Play version, 3.25.0 RC1, 3.19.1, but no dice on any of these versions. I've tried toggling permissions, but nothing gets uploaded - neither old or new photos. It does not work at all :(

**Comment by Blisk** [comment]
I still have this problem with 26.0.2 and cant solve this.

**Comment by Taoshan98** [comment]
I also had the same problem, I don't know how I solved it, I will list the steps I did:

- Remove permissions and give them back to the app.
- Create a 'Camera' folder and enable automatic synchronization

at this point nothing was working yet, then I to try I downloaded the desktop client to my Mac and there I started syncing everything I had manually uploaded to the server.

then I deleted the 'Camera' folder that I had created previously and suddenly I saw an 'InstantUpload' folder appear among the folders with the 'Camera' folder inside and it started automatically uploading everything

I don't know how useful this comment can be but it is to share my experience.

**Comment by JeCheeseSmith** [comment]
I have been troubleshooting with this a few days now. I tried granting permissions again, re-syncing (turning cloud logo on/of), that didn't work. (And ofc restarting my phone and things)

Found:
- I figured it did only upload certain photos, not all of them, which might have to do with a certain time/only newly created files.
- Taking a photo does re trigger the auto upload for my Camera Photo album, however i have different folders in which photos are stored, which still won't be uploaded. 

I later thought it was a problem with the suggested folders so I set up custom folders for my files, photos, videos and other media. - That does work.

However, after a few hours now, the app would randomly terminated. (Which I think could be somehow caused by the OS? Or maybe the way the app works? (I haven't looked into the code) ) Battery optimization is turned for Nextcloud (as needed).

Edit:
I also just found some replies on this post https://www.reddit.com/r/NextCloud/comments/telrfy/auto_upload_files_does_not_work_anymore/ , saying to go to Accounts and enable sync. I don't know if this does anything yet, it was turned off so I enabled it. 

Question: Would it maybe be usefull to have some sort of post/FAQ page for people to be present with the given "workarounds"? (I found a lot of posts, articles & reddit pages about the AutoUpload features)

**Comment by biva** [comment]
2 things I did and it works now, I don't know for how long it will work though:
1. Go to Settings > Automatic upload > Press the 3 dots on the top right > Deactivate battery optimiser (sorry if the translation is not exact)
2. I had an old installation of Nextcloud on Android, with the data in a different location → I removed this old folder
Good luck!

**Comment by maranov** [comment]
Seems like an issue with the more fine-grained file permissions on latest Androids. Giving the Nextcloud app an **All files access** (Settings -> Apps -> Special app access) seems to fix the issue.

Android 13, Moto G73, NC 3.25.0 via GPlay.

**Comment by 007hacky007** [comment]
> Seems like an issue with the more fine-grained file permissions on latest Androids.

Probably not limited just to this. Because on older Androids (Android 9) there are no fine-grained permissions and it also does not work there.

Older version 3.19.1 works fine for me though.

**Comment by BLu85** [comment]
`v3.25.0`: AutoUpload for Camera and Video folders does not work.
The Camera folder, on both local and server side, has 6k+ pictures. On the local folder there are some more pictures and these should be detected and auto uploaded, but this does not happen.

I tried the workarounds posted here (remove and give storage permission to the Nextcloud app, activate the "sync files" in settings - account - nextcloud, create a custom folder, take a pic, rename the remote folder). App is unusable as it also consumes a lot of battery (God knows what it is doing in the background, rather than just stay in idle waiting for the charge to be connected, as it is set in the Auto Upload settings).
This issue on the AutoUpload feature has been around for months if not years. It would be nice to have it fixed. At the end of the day, is it not the main task this app should do (i.e. auto upload and keep files in sync)?

Any help or suggestion is appreciated.
Thanks

P.s.: I tried to set up a tmp folder with few pics and those got detected and auto uploaded correctly. Could it be that the App gives up when there are too many pics in a folder?

**Comment by Osmodia666** [comment]
> `v3.25.0`: AutoUpload for Camera and Video folders does not work. The Camera folder, on both local and server side, has 6k+ pictures. On the local folder there are some more pictures and these should be detected and auto uploaded, but this does not happen.
> 
> I tried the workarounds posted here (remove and give storage permission to the Nextcloud app, activate the "sync files" in settings - account - nextcloud, create a custom folder, take a pic, rename the remote folder). App is unusable as it also consumes a lot of battery (God knows what it is doing in the background, rather than just stay in idle waiting for the charge to be connected, as it is set in the Auto Upload settings). This issue on the AutoUpload feature has been around for months if not years. It would be nice to have it fixed. At the end of the day, is it not the main task this app should do (i.e. auto upload and keep files in sync)?
> 
> Any help or suggestion is appreciated. Thanks
> 
> P.s.: I tried to set up a tmp folder with few pics and those got detected and auto uploaded correctly. Could it be that the App gives up when there are too many pics in a folder?



Same Problem here. Upload works in 1 of 100 cases. Do the devs even care? This Problem is around for Months, or even Years?

**Comment by NicolasGoeddel** [comment]
> This Problem is around for Months, or even Years?
It's years!

I already switched to FolderSync which works totally fine with syncing data to Nextcloud. The only thing that does not work is to automatically create a folder structure (year/month) from the modified/created data extracted from the pictures' and videos' EXIF data which is a bummer, because I really like that functionality and seemingly only the Nextcloud app itself has that feature.

**Comment by 007hacky007** [comment]
> I'm done with this app definitely. I have tried 2 weeks to find a fix for it to make it work. Dear developers, go to McDonald's!

Did you try older version as suggested in my previous post? 3.19.1 works fine on both old Android 9 and on newest Android. It's syncing photos just fine...

**Comment by NicolasGoeddel** [comment]
> Did you try older version as suggested in my previous post? 3.19.1 works fine on both old Android 9 and on newest Android. It's syncing photos just fine...

Downgrades are not supported by Google Play.

**Comment by 007hacky007** [comment]
> > Did you try older version as suggested in my previous post? 3.19.1 works fine on both old Android 9 and on newest Android. It's syncing photos just fine...
> 
> Downgrades are not supported by Google Play.

Uh. Just uninstall your current version, manually download apk of the 3.19.1 release here https://github.com/nextcloud/android/releases/tag/stable-3.19.1 and install it.

**Comment by Blisk** [comment]
I noticed that Nextcloud is the biggest consumer of battery it is top on the list. It drains the battery very fast.

**Comment by joshtrichards** [comment]
I'm going to close this issue for the following reasons:

- The original reporter no longer has this situation (see https://github.com/nextcloud/android/issues/9320#issuecomment-999834785)
- It has become a "catch-all" for various auto-upload problems (which makes is near impossible as a developer to sort though which ones are relevant to the reported issue... and which are entirely unrelated and should have their own issue)
- It's been two years and numerous code changes have occurred so it's just as likely we'll be chasing things that aren't even relevant today
- Several solutions have been posted that address various vendor specific scenarios
- Various enhancements have been made surrounding permissions management

### If auto-uploads are not being triggered for you, please open a dedicated issue with the complete info about your situation. 

Also, please check your *Uploads* tab to see the status of uploads. Ones queued but blocked for power reasons will show up there, for example. If your media files aren't even showing up in the queue at all that's also of interest, but an entirely different root cause most likely. Again, dedicated issues and let's work through thing one-by-one. 

In addition, if you want to be even more helpful:

* provide debug logs (https://github.com/nextcloud/android/#getting-debug-info-via-logcat-mag)
* try to reproduce in a clean install of the Dev version
  - you can install a Dev app version in parallel w/o impacting your main app installation (it'll have it own independent device storage and you can enable auto-upload in it *only* for a test folder)

Thank you all for your feedback. I know dealing with and reporting bugs takes time and energy- often just as much as tracking down the cause and fixing them does.

**Comment by BLu85** [comment]
Hi @joshtrichards 

Thanks for commenting on this issue.

I think a couple of reasons for this thread to be so silent is because some people have switched to using other apps (FolderSync) or because have stopped using Nextcloud App (like myself). Furthermore, only 2 versions have been released in 1 year, hence I am still waiting for the next one to be released before commenting again saying if it has sorted or not the AutoUpload.

The AutoUpload is an essential feaure for this App, not optional (like the image cropping for example). It should always work. In my specific case: I have 4 Nextcloud users. The AutoUpload feature works well on 2 phones, it doesn't work on the other 2. Settings are the same.

I haven't seen many comments from the developers on this thread asking for debug help or general info to tackle the problem. I am sure they have been working on it, but until now we (the users on this thread) haven't been involved much. We are all happy to contribute to solve it.

You suggested, for those still affected from this problem, to open a dedicated issue about the AutoUpload. Mine was opened months ago and still hasn't seen an answer from the devs. Here it is #11741. I personally don't think that this proliferation of tickets will help to sort the problem. But if this is the way to gather the attention of the devs to tacke and solve it, then I'll open one more.
I really hope that this feature will be solved once for all so that we can go back at happily using our Nextcloud App.

Thanks,
Luca

**Comment by Blisk** [comment]
As I see now, the major problem is that Nextcloud for Android is the biggest consumer of battery. I get constantly message that Nextcloud use a battery and it is on the top of battery drainers. And then it kills nextcloud service.
I don't get that from Snapchat or Viber or anything like that. So it must be a bad app for Android, nothing else.

**Comment by Dedsd** [comment]
> is your camera folder on an SD card by any chance?
> 
> I had the same problem last month the last time I posted. I suspect it was an issue related to sd card permissions or something weird happening with android not granting permission for the sd card cause other folder on the internal memory were auto uploading.
> 
> I resolved it by installing an new sd card setup auto-upload with the new sd card took a photo to ensure it would upload. Then I Removed the new sd card and re-inserted the old one, setup auto-upload again and it started working. I am still uncertain why it stopped working and why it would not start.

Hello @meldarionqeusse 

I'm using an micro sd card, what did you do?

**Comment by meldarionqeusse** [comment]
As I explained I tried auto upload with a new SD card and it worked. I suspect cause the new SD card had few photos on it only. Then I re inserted my old SD card which has 3 years of photos on it configured folders again on nextcloud and after while it started syncing.

I think if you have lots of photos it might take a an hour. Also make sure you don't have battery saving modes enabled. I noticed that my phone was not syncing as soon as I take a photo but would start syncing a few minutes after I plug it in at night to charge cause stamina mode gets disabled.

**Comment by poltpolt** [comment]
NC 28 and years later the bug still exists on all my family members phones, what a shame, i always admired nextcloud but having this bug present for years is just shameful

**Comment by MarizanneVis** [comment]
Also just adding on to this as the issue is still present for me after implementing every fix I could find online and in this thread.
Also adding, FolderSync and Syncthing really aren't good alternatives because they 1) can't sync the root directory of the phone, and 2) can't detect media folders like nextcloud, so it's a massive task to try and backup all folders with media for every client.

This really is the single feature I still need from NextCloud to allow me to get off Google.

**Comment by mnalis** [comment]
@MarizanneVis (and everyone else adding _"me too"_s) it would be much more helpful if you listed **_exactly_** what you **_tried_** (if accompanied with screenshots or screencasts, even better!), and **_exactly_** was the **_result_** was for each of the things you tried.  Also, which versions of NC, on server and client, as well as Android version (and manufacturer/model - because of [manufacturer specific issues](https://dontkillmyapp.com/)) would be extremely helpful. Ideally by filing new bug template and adding link to this similar issue (instead of adding comment to closed issue).

Saying _"I tried all fixes I could find"_ and _"They didn't work"_ is a school example of how **not** to report bugs.
Nice background article on how and why: ["How to Report Bugs Effectively"](https://www.chiark.greenend.org.uk/~sgtatham/bugs.html).

Please note that I'm not saying that to be mean to you :heart: (nor do I target you specifically here - it applies to everyone, including myself!), but to improve chances of bug actually being determined and solved (or workaround provided). 

**Without good quality bug reports** full of specific data (which do take some time and effort to write, I know!) there is probably **no chance of the bug ever being** successfully reproduced, much less **fixed**. And only users with this elusive problem can provide such bug reports. (Note that even with good bug reports there is guarantee that the bug will be fixed, but it is a prerequisite for a chance of it being fixed - and probably required several different good bug reports to isolate unrelated variables).

**Comment by LorenzoVonMatterhom** [comment]
Just for reference this happened to me and since turning on off various settings didn't work I had to delete archive and cache of the application. Logging in again apparently worked (for now)

**Comment by metroengines-vlad** [comment]
> Just for reference this happened to me and since turning on off various settings didn't work I had to delete archive and cache of the application. Logging in again apparently worked (for now)

Exact same situation. Phone: Oppo Reno 5A; latest version of the android app 3.30.6. Nextcloud Hub 8 29.0.7.

I suspect the most recent update to the app broke the Auto-uploads' previous settings as there was some sort of a permission change from Android side: we no longer have the option to remove files on the phone after upload, so Nextcloud devs had to disable that option... and potentially previous settings became invalid/corrupted?

Anyway, clearing out the App Settings --> re-logging in (and then setting up the autouploads from a clean state) worked for me as well.

**Comment by meldarionqeusse** [comment]
Yes I can confirm the most recent update of the android app broke auto upload. Disabling auto uoload for the folders by Clicking on the cloud icon and re-enabling it made things work again for new. However old photos did not automatically sync I had to upload them manually.

This is an issue because the upload happened in the background and I did not realize the feature stopped working. The timing also was very unfortunate since it holiday season lots of photos taken and changing to new phone so I was going to lose a bunch of photos. 

I really wish people stop pushing updates during the holiday season and breaking things unless it a security update

**Comment by ados8** [comment]
Auto upload barely works still, why is this closed.
If I toggle off and on the cloud upload for that gallery it sometimes will sync a few.
I've completely reset the app, uninstalled and reinstalled and it synced the missing photos and then new ones did nothing.

**Comment by Blisk** [comment]
I also can confirm that version 29 stopped to upload my phone pictures on nextcloud. After I update it to 30 than it start to work again.

**Comment by wwebers** [comment]
And it stopped again after updating to 31. This seem to be a never ending story...

**Comment by Norde** [comment]
> And it stopped again after updating to 31. This seem to be a never ending story...

Same here, and this time the trick of [https://github.com/nextcloud/android/issues/9320#issuecomment-991918596](@martinmoreira) is not working.

Using the F-droid version so not related to issue #14409 

Any update or plan regarding this big issue ?

**Comment by newachu** [comment]
For me, uninstalling Nextcloud app, then reinstalling it, logging into my provider and granting permissions to storage and running in the background worked. Auto upload started working after configuring it in Nextcloud - settings - sync - auto upload. Even the feature to delete the original media after uploading was re-enabled.

What didn't work: Before uninstalling, tweaking the existing file access permissions off and on. Re-enabling the account sync in phone settings - users and accounts. Rebooting the phone. Reconfiguring Nextcloud settings - auto upload.

My suspicion is that this relates to having the Nextcloud app installed for a long time in the phone, and going through several android updates and android permission model changes. Which would also make it difficult to triage, since installing the app for the first time in the phone just works.

**Comment by NicolasGoeddel** [comment]
The thing is: Nobody wants to completely uninstall and reinstall the Nextcloud app when the upload configuration can not be backed up. Last time I used the app for automatic uploads I had a lot of different folder configured properly and with a reinstallation everything was gone and there was no way to backup all the settings and restore them later.
That's the reason why I don't use the Nextcloud app anymore for any uploads. It just doesn't work reliably. And even when I want to upload a single file using the share functionality it always only works on the second try and I don't know why. It's just bad and since years now it was never solved properly it seems.
That's why I use syncthing now to just sync my whole phone with my server. I syncs within seconds and is very reliable. And it even works uni-directional which I like a lot.

**Comment by AxelPanningDE** [comment]
I have the same issues. I have one interesting observation though. It appears that always maximum ONE sych setup works. E.g. i get running DCIM/Camera while DCIM/Screenshots does not. Once i get (by whatever triggers /reset) Screenshots working, then immediately the Camera folder doesnt work anymore. 

Seems there is some magic single token that can be utilized only.

**Comment by alperozturk96** [comment]
Hello

We created a [pinned issue](https://github.com/nextcloud/android/issues/16550) for that. Could you please download latest APK given in the issue and share the logs with us?
